src/webui/webapplication.cpp

```diff
@@ -320,7 +346,9 @@ void WebApplication::action_command_download()
             if ((url.size() == 40 && !url.contains(QRegExp("[^0-9A-Fa-f]")))
                 || (url.size() == 32 && !url.contains(QRegExp("[^2-7A-Za-z]"))))
                 url = "magnet:?xt=urn:btih:" + url;
-            BitTorrent::Session::instance()->addTorrent(url);
+
+            Net::DownloadManager::instance()->setCookiesFromUrl(cookies, QUrl::fromEncoded(url.toUtf8()));
```

Comment:
You can't.  You need to set the cookie per URL and the url variable is inside the loop.  Remember you can download several URLs at the same time.